### PR TITLE
Enable strong mode, a few linter rules and run dartfmt on most files

### PR DESCRIPTION
**Pull request by isoos**
Apologies for the long PR, the changes are mostly non-functional, with the exception of calling `.toString()` in `app/bin/tools_remove_package.dart` line 95, and using `dart:math`'s `min` in `app/lib/search_service.dart`.

They started out as separate commits, but got squashed along the way, and it would have been too painful to get separated again.

### CODE REVIEW

File: app/lib/atom_feed.dart [hunk 1]
```diff
@@ -23,25 +23,25 @@ class FeedEntry {
   final String alternateTitle;
 
   FeedEntry(this.id, this.title, this.updated, this.authors, this.content,
-            this.alternateUrl, this.alternateTitle);
+      this.alternateUrl, this.alternateTitle);
```
Comment (mkustermann) [inline app/lib/atom_feed.dart]: There is another place further up, where you replaced a `Configuration.prod() ...;` with `Configuration.prod() ... {}`. But here you don't add an empty clause.

I guess it means if there are field initializes **only then** do we also need the empty block?

Comment (isoos) [inline app/lib/atom_feed.dart]: I believe I have only removed `{}` (and replaced them with `;`).

Comment (mkustermann) [inline app/lib/atom_feed.dart]: Oh, sorry. Got probably irritated by this github diff showing tool. We use normally rietveld which is much nicer for reviewing.

**Comment by mkustermann** [comment]
You could update `README.md` how to run the analyzer on the package if you like.

**Pull request merged**